tools/building.py

```diff
@@ -1298,8 +1298,23 @@ def is_bitcode(filename):
 
 
 def is_wasm(filename):
-  magic = open(filename, 'rb').read(4)
-  return magic == b'\0asm'
+  if not os.path.exists(filename):
+    return False
+  header = open(filename, 'rb').read(webassembly.HEADER_SIZE)
+  return header == webassembly.MAGIC + webassembly.VERSION
+
+
+def is_wasm_dylib(filename):
+  """Detech wasm dynamic libraries by the presence of the "dylink" custom section."""
+  if not is_wasm(filename):
+    return False
+  module = webassembly.Module(filename)
+  section = next(module.sections())
+  if section.type == webassembly.SecType.CUSTOM:
+    module.seek(section.offset)
```

Comment:
I guess this is pre-existing, but I wonder if there would be a way to avoid having to manually seek the module after we've iterated through the sections this way; it seems redundant. Maybe something to think about for future improvements.

Reply:
oh actually the generator does `self.seek(offset)`. Does that mean this seek isn't needed?

Reply:
I think its best not to assume to position that the iterator left the file in.

I have thought about the efficiency of all the seek() and my thoughts so far are:

1. Lets measure once we start using this code for more stuff
2. Seeks should be very cheap in the OS / libc / python 
3. Is seeks aren't cheep we can just call `os.mmap` after open which would then make them free (I think).

Reply:
Yeah it's true that seeking is generally cheap. And especially if it's redundant anyway :)